src/runtime/frame.cpp

```diff
@@ -347,10 +347,19 @@ extern "C" PyFrameObject* PyFrame_New(PyThreadState* tstate, PyCodeObject* code,
 extern "C" BORROWED(PyObject*) PyFrame_GetGlobals(PyFrameObject* f) noexcept {
     return BoxedFrame::globals((Box*)f, NULL);
 }
+
 extern "C" BORROWED(PyObject*) PyFrame_GetCode(PyFrameObject* f) noexcept {
     return BoxedFrame::code((Box*)f, NULL);
 }
 
+extern "C" BORROWED(PyObject*) PyFrame_GetBack(PyFrameObject* f) noexcept {
+    return BoxedFrame::back((Box*)f, NULL);
+}
+
+extern "C" void PyFrame_SetBack(PyFrameObject* f, PyFrameObject* back) noexcept {
+    BoxedFrame::back((Box*)f, back);
```

Comment:
I don't think this does what you want it to do -- it just fetches the current f_back.  We don't currently have a way to set f_back; we could add one, but it would be a bit complicated (I think we would need to update `frame_info->back` as well?) and we would need to test it really well.

I spent some time looking at the Cython code that is calling this though, and as far as I can tell, it is a hacky and buggy way of implementing a feature that shouldn't exist :P .  I am ok with us just using the if-PyPy side which does nothing, which I think would be saner and also not require us to add PyFrame_SetBack.  If we really want to support that particular Cython code, it might be better to add a higher-level API since I don't know if we can always support PyFrame_SetBack properly.